diff --git a/boards/apollo3_evb/examples/hello_world/gcc/startup_gcc.c b/boards/apollo3_evb/examples/hello_world/gcc/startup_gcc.c
--- a/boards/apollo3_evb/examples/hello_world/gcc/startup_gcc.c
+++ b/boards/apollo3_evb/examples/hello_world/gcc/startup_gcc.c
@@ -111,7 +111,7 @@
 //
 //*****************************************************************************
 __attribute__ ((section(".stack")))
-static uint32_t g_pui32Stack[1024];
+static uint32_t g_pui32Stack[1024*20];
 
 //*****************************************************************************
 //
